### DISCUSSION TITLE
Admin authentication

### DISCUSSION THREAD

**Discussion by reisfmb** [pocketbase/pocketbase#4318] [Q&A] [answered]
Can someone provide me an example of how admin authentication works with any SSR framework?

The usual users collection authentication works fine, following those examples
- https://github.com/pocketbase/js-sdk?tab=readme-ov-file#ssr-integration
- https://pocketbase.io/docs/authentication/#authenticate-as-app-user

The thing is that idk how to adapt it to work with admins instead of the users collection.

Here's what I've tried:
1. change the ssr integration to use `pb.admins.client.authStore` and `pb.admins.authRefresh`;
2. used [authenticate-as-admin](https://pocketbase.io/docs/authentication/#authenticate-as-admin) reference on the login service.

Without success here's what happens with those attempts:
- `pb.admins.authWithPassword` raise the following error: _"The request requires valid admin authorization token to be set."_;
- the js-sdk `exportToCookie` helper generates token that looks incorrect and have an expiration date < now, therefore they are being passed on further requests;
- This is an example of a token retrieved by that function: `pb_auth=%7B%22token%22%3A%22%22%2C%22model%22%3Anull%7D; Path=/; Expires=Thu, 01 Jan 1970 00:00:00 GMT; HttpOnly; SameSite=Strict`

I'm pretty sure I'm doing something wrong or there are steps I haven't implemented yet... would appreciate any help on this :pray: 

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
When using the PocketBase JS SDK as admin it is not required to have a unique instance for each request and it is safe to have a single global admin client that you can just import whenever you want to make an admin request (aka. you don't have to deal with cookies).

I'll try to find time to update the docs in the near future, but for now you can check the example from https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7621130.

**Reply by reisfmb** [reply]
Thanks for replying @ganigeorgiev !

I keep getting this when I call `pb.admins.authWithPassword` (commented out the backend url with <...>):
```bash
url: '<...>',
  status: 401,
  response: {
    code: 401,
    message: 'The request requires valid admin authorization token to be set.',
    data: {}
  },
  isAbort: false,
  originalError: {
    url: '<...>',
    status: 401,
    data: {
      code: 401,
      message: 'The request requires valid admin authorization token to be set.',
      data: {}
    }
  }
```

**Reply by ganigeorgiev** [reply]
The error code doesn't make sense as the `api/admins/auth-with-password` action is public. Something else is going on but there are not enough details to help you debug it.

Please check again the example linked in https://github.com/pocketbase/pocketbase/discussions/3792#discussioncomment-7621130.

If that doesn't help, then please provide a code sample showing what you are doing.

**Comment by orenaksakal** [comment]
This is what I use for my sveltekit webhooks

the admin client

```javascript
// pocketbaseAdminClient.ts
import { SECRET_POCKETBASE_EMAIL, SECRET_POCKETBASE_PW } from '$env/static/private';
import { PUBLIC_POCKETBASE_URL } from '$env/static/public';
import type { TypedPocketBase } from '$lib/types/pocketbase-types';
import PocketBase from 'pocketbase';

export const pocketbaseAdminClient = new PocketBase(PUBLIC_POCKETBASE_URL) as TypedPocketBase;

pocketbaseAdminClient.autoCancellation(false);

// the credentials can be filled via env variables
await pocketbaseAdminClient.admins.authWithPassword(SECRET_POCKETBASE_EMAIL, SECRET_POCKETBASE_PW, {
	// This will trigger auto refresh or auto reauthentication in case
	// the token has expired or is going to expire in the next 30 minutes.
	autoRefreshThreshold: 30 * 60
});
```

and usage on webhook route

```javascript
// my/webhook/route/+server.ts
import { error, json } from '@sveltejs/kit';
import type { RequestHandler } from './$types';
import { Collections, type RendersResponse, type UsersResponse } from '$lib/types/pocketbase-types';

import { pocketbaseAdminClient } from '$lib/pocketbase/adminClient';

export const POST: RequestHandler = async ({ request, params }) => {
	const body = await request.json();
	const formData = new FormData();

	formData.append('completed', 'true');

	await pocketbaseAdminClient.collection(Collections.MyCollection).update(params.id, formData);

	return json({ message: 'success' });
};
```

**Comment by reisfmb** [comment]
Ok, I figured it out... I was passing `http` instead of `https` when instantiating the `PocketBase` class as the link of my hosted pocketbase application.

Thanks a lot @ganigeorgiev and @orenaksakal
